src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,3047 @@
+            env.close();
+
+            std::uint32_t aliceTicketSeq{env.seq(alice) + 1};
+            env(ticket::create(alice, 10));
+            env.close();
+
+            auto const aliceSeq = env.seq(alice);
+            auto const batchFee = batch::calcBatchFee(env, 0, 2);
+            uint256 const chkId{getCheckIndex(alice, aliceSeq)};
+            auto const batchTxn = env.jt(
+                batch::outer(alice, 0, batchFee, tfAllOrNothing),
+                batch::inner(check::create(alice, bob, XRP(10)), aliceSeq),
+                batch::inner(pay(alice, bob, XRP(1)), 0, aliceTicketSeq + 1),
+                ticket::use(aliceTicketSeq));
+            auto const objTxn = env.jt(check::cash(bob, chkId, XRP(10)));
+            env(batchTxn);
+            env(objTxn, ter(tecNO_ENTRY));
+            env.close();
+
+            auto const txIDs = batchTxn.stx->getBatchTransactionIDs();
+            TxID const parentBatchId = batchTxn.stx->getTransactionID();
+            Json::Value const jrr = getLastLedger(env);
+            std::vector<TestLedgerData> testCases = {
+                {0, "Batch", "tesSUCCESS", to_string(parentBatchId), false},
+                {1, "CheckCreate", "tesSUCCESS", to_string(txIDs[0]), true},
+                {2, "Payment", "tesSUCCESS", to_string(txIDs[1]), true},
+                {3,
+                 "CheckCash",
+                 "tesSUCCESS",
+                 to_string(objTxn.stx->getTransactionID()),
+                 false},
+            };
+            validateLedgerTxns(env, jrr, testCases, parentBatchId);
+        }
+    }
+
+    void
+    testPseudoTxn(FeatureBitset features)
+    {
+        testcase("pseudo txn");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        test::jtx::Env env{*this, envconfig()};
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        env.fund(XRP(1000), alice, bob);
+        env.close();
+
+        STTx const stx = STTx(ttAMENDMENT, [&](auto& obj) {
+            obj.setAccountID(sfAccount, AccountID());
+            obj.setFieldH256(sfAmendment, uint256(2));
+            obj.setFieldU32(sfLedgerSequence, env.seq(alice));
+            obj.setFieldU32(sfFlags, tfInnerBatchTxn);
+        });
+
+        std::string reason;
+        BEAST_EXPECT(isPseudoTx(stx));
+        BEAST_EXPECT(!passesLocalChecks(stx, reason));
+        BEAST_EXPECT(reason == "Cannot submit pseudo transactions.");
+        env.app().openLedger().modify([&](OpenView& view, beast::Journal j) {
+            auto const result = ripple::apply(env.app(), view, stx, tapNONE, j);
+            BEAST_EXPECT(!result.applied && result.ter == temINVALID_FLAG);
+            return result.applied;
+        });
+    }
+
+    void
+    testBatchWithSelfSubmit(FeatureBitset features)
+    {
+        testcase("batch with self submit");
+        // IMPORTANT: When a transaction is self submitted and another
```

Comment:
What exactly do you mean by self submit?

Reply:
"self submit" was just referring to when the first batch transaction is also the submitter of the batch outer. If you would like other nomenclature to be used please feel free to add a suggestion. The test itself probably relies more on the open ledger and the ordering of transactions more than the "self submit" nature.